cns/restserver/api_test.go

```diff
@@ -623,14 +631,18 @@ func TestGetNetworkContainerVersionStatus(t *testing.T) {
 		podNamespace: "testpodnamespace",
 	}
 
-	mnma.GetNCVersionF = func(_ context.Context, _ nmagent.NCVersionRequest) (nmagent.NCVersion, error) {
-		return nmagent.NCVersion{}, errors.New("boom") //nolint:goerr113 // it's just a test
+	mnma.GetNCVersionListF = func(_ context.Context) (nmagent.NCVersionList, error) {
+		rsp := nmagent.NCVersionList{
+			Containers: []nmagent.NCVersion{},
+		}
+		return rsp, errTest
 	}
+
 	mnma.JoinNetworkF = func(_ context.Context, _ nmagent.JoinNetworkRequest) error {
-		return errors.New("boom") //nolint:goerr113 // it's just a test
```

Comment:
These should be reverted back. The point of these dynamic errors is that the actual error is completely inconsequential. At the same time, we want to make it obvious that we definitely _intend_ to return an error here. Assigning it to a package variable does a few things opposing this:

1) I'm not really sure what `errTest` actually is: it's typed as an error, but is it nil? Did someone else assign to it? I don't know.
2) Because there is no way to make it immutable, it is inter-test state that can create subtle flakiness as these tests change over time.
3) It gives the error an identity that we want to avoid. `JoinNetwork` can fail in any number of ways, and we want to say here that the specific error is completely inconsequential (only that it implements `error`). This is to encompass timeouts, bad requests, actual failures, etc.

So yeah, even though the linter complains about this, the linter is wrong.

Reply:
I think we can leave it as it is for this PR and I can open another PR to address this issue.

Reply:
@csfmomo, @timraymond is asking that you _not_ make the change here, so please revert it. No need to change it, then open another PR to change it back.

Reply:
Reverted.